web/js/mapUI/components/tile-errors/tile-error-handler.js

```diff
@@ -0,0 +1,171 @@
+import { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { useSelector, useDispatch } from 'react-redux';
+import { clearErrorTiles as clearErrorTilesAction } from '../../../modules/ui/actions';
+import {
+  selectDate as selectDateAction,
+  selectInterval as selectIntervalAction,
+} from '../../../modules/date/actions';
+import { getNextDateTime } from '../../../modules/date/util';
+import { subdailyLayersActive } from '../../../modules/layers/selectors';
+
+// updating timezone for subdaily times
+function convertTimestamp(timestamp) {
+  const date = new Date(timestamp);
+  // add 4 hours to the time
+  date.setHours(date.getHours() + 4);
+  // set the seconds to 0
+  date.setSeconds(0);
+  // round down the minutes to the nearest multiple of 10
+  const roundedMinutes = Math.floor(date.getMinutes() / 10) * 10;
+  // format the updated date back to a string
+  const year = date.getFullYear();
+  const month = String(date.getMonth() + 1).padStart(2, '0');
+  const day = String(date.getDate()).padStart(2, '0');
+  const hours = String(date.getHours()).padStart(2, '0');
+  const minutes = String(roundedMinutes).padStart(2, '0');
+
+  return `${year}-${month}-${day}T${hours}:${minutes}:00`;
+}
+
+function formatDate(dateString, hasSubdailyLayers) {
+  const date = new Date(dateString);
+  const year = date.getFullYear();
+  const month = String(date.getMonth() + 1).padStart(2, '0');
+  const day = String(date.getDate()).padStart(2, '0');
+  const hours = String(date.getHours()).padStart(2, '0');
+  const minutes = String(date.getMinutes()).padStart(2, '0');
+  const seconds = String(date.getSeconds()).padStart(2, '0');
+
+  if (hasSubdailyLayers) {
+    const timestamp = `${year}-${month}-${day}T${hours}:${minutes}:${seconds}`;
+    const adjustedTimezone = convertTimestamp(timestamp);
+    return adjustedTimezone;
+  }
+
+  return `${year}-${month}-${day}T00:00:00`;
+}
+
+// gets a week ago from real time date
+function weekAgo(realTimeDate) {
+  const inputDate = new Date(realTimeDate);
+  const oneWeekInMilliseconds = 7 * 24 * 60 * 60 * 1000;
+  const earlierDate = new Date(inputDate.getTime() - oneWeekInMilliseconds);
+  const earlierDateString = earlierDate.toString();
+  return earlierDateString;
+}
+
+// safe gaurd agaisnt stepping back too far, we only want to go back 7 days for daily and 24 hours for subdaily
```

Comment:
I like your comments overall but I would remove the reference to subdaily here as it is handled directly below & contradicts between 23/24 hours.